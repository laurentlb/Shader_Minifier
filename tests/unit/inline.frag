void main()
{
  float x = 0.5;
  float i_y = 0.6*x;
  float a = x * i_y;
}

int arithmetic()
{
  int i_a = 2;
  int i_b = 3;
  int i_c = i_a + i_b;
  return 4 * i_a * i_c;
}

int vars(int arg, int arg2)
{
  int i_a = arg;
  int i_b = arg2;
  int i_c = i_a + i_b;
  return i_a * i_c;
}
