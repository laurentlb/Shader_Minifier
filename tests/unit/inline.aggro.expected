#version 120

float result;
void main()
{
  result=.15;
}
int arithmetic()
{
  return 40;
}
int vars(int arg,int arg2)
{
  return arg*(arg+arg2);
}
int arithmetic2()
{
  return 40;
}
int unusedVars()
{
  return 30;
}
int unusedVars2()
{
  return 18;
}
int multiPass()
{
  return 3;
}
float multiPass2()
{
  return 9.;
}
int dont_inline_lvalue()
{
  int a=1;
  a=2;
  return 3;
}
float arr[]=float[](3.4,4.2);
void lvalues()
{
  arr[1]=2.;
}
uniform int time;
in int sync;
int dependOnConst()
{
  return(time+sync)*2*3;
}
